runtime/src/builtins.rs

```diff
@@ -14,13 +14,21 @@ use solana_frozen_abi::abi_example::AbiExample;
 fn process_instruction_with_program_logging(
     process_instruction: ProcessInstructionWithContext,
     program_id: &Pubkey,
+    first_instruction_account: usize,
     instruction_data: &[u8],
     invoke_context: &mut dyn InvokeContext,
 ) -> Result<(), InstructionError> {
+    debug_assert_eq!(first_instruction_account, 1);
```

Comment:
@Lichtso it's possible to hit this debug assert with the tx I described here: https://github.com/solana-labs/solana/pull/21238/files?diff=split&w=0#r752263887